VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/SwitchDoesNotHandleAllEnumOptions/SwitchDoesNotHandleAllEnumOptionsAnalyzer.cs

```diff
@@ -26,24 +28,63 @@ internal static DiagnosticDescriptor Rule
 
         private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
         {
-            var switchBlock = (SwitchStatementSyntax) context.Node;
+            var switchBlock = (SwitchStatementSyntax)context.Node;
 
             var enumType = context.SemanticModel.GetTypeInfo(switchBlock.Expression).Type as INamedTypeSymbol;
             if (enumType == null || enumType.TypeKind != TypeKind.Enum)
             {
                 return;
             }
 
-            var caseLabels = switchBlock.Sections.SelectMany(l => l.Labels)
-                    .OfType<CaseSwitchLabelSyntax>()
-                    .Select(l => l.Value)
-                    .ToList();
+            var caseLabels = new List<ExpressionSyntax>();
 
-            var labelSymbols = caseLabels.Select(l => context.SemanticModel.GetSymbolInfo(l).Symbol);
+            foreach (var section in switchBlock.Sections)
+            {
+                foreach (var label in section.Labels)
+                {
+                    if (label.IsKind(SyntaxKind.CaseSwitchLabel))
+                    {
+                        caseLabels.Add(((CaseSwitchLabelSyntax)label).Value);
+                    }
+                }
+            }
+
+            var labelSymbols = new List<ISymbol>();
+            foreach (var label in caseLabels)
+            {
+                var symbol = context.SemanticModel.GetSymbolInfo(label).Symbol;
+                if (symbol == null)
+                {
+                    // potentially malformed case statement
+                    // or an integer being cast to an enum type
+                    return;
```

Comment:
Oh right. The `return` statement also breaks out of the encompassing method. Alright, all good then I think.